src/world_new/World.cpp

```diff
@@ -0,0 +1,76 @@
+//
+// Created by john on 12/16/19.
+//
+
+#include "world_new/World.hpp"
+
+#include <utility>
+
+#include "Settings/Settings.h"
+#include "world_new/views/WorldDataView.hpp"
+
+namespace rtt::world_new {
+WorldData const &World::setWorld(WorldData &newWorld) noexcept {
+    if (currentWorld) {
+        toHistory(currentWorld.value());
+    }
+    currentWorld = std::move(newWorld);
+    return currentWorld.value();
+}
+
+void World::toHistory(WorldData &world) noexcept {
+    updateTickTime();
+    if (history.size() < HISTORY_SIZE && currentIndex < history.size()) {
+        history.emplace_back(std::move(world));
+    } else {
+        history[currentIndex] = std::move(world);
+        currentIndex %= HISTORY_SIZE;
+    }
+    currentIndex++;
+}
+
+std::optional<view::WorldDataView> World::getWorld() const noexcept {
+    if (currentWorld) {
+        return view::WorldDataView{&*currentWorld};
```

Comment:
You are using the overloaded dereference operator here right? Maybe you can explain briefly what you do here above the line, I can imagine this is weird for most people